kernel/src/log_segment.rs

```diff
@@ -112,13 +121,234 @@ impl LogSegment {
     }
 }
 
+pub(crate) struct LogSegmentBuilder<'a> {
```

Comment:
nit: docs; can be something short/simple like "builder for log segment from [start_version, end_version] _inclusive_!"